### DISCUSSION TITLE
Use exposeProxy instead of proxy for public component methods?

### DISCUSSION THREAD

**Discussion by Dtsiantaris** [quasarframework/quasar#17346] [Ideas / Proposals]
Maybe I am wrong here not knowing what could actually break here if `exposeProxy` was used instead of `proxy`. 
The public methods are now being exposed like this(QBtn for reference):

```
setup (props, { slots, emit }) {
    const { proxy } = getCurrentInstance()

 // expose public methods
    Object.assign(proxy, { click: onClick })
```

But shouldn't it be this:

```
setup (props, { slots, emit }) {
    const { exposeProxy, proxy } = getCurrentInstance()

 // expose public methods
    Object.assign(exposeProxy, { click: onClick })
// use proxy like before wherever it's needed
```

I might have misunderstood something but wouldn't this way be the equivalent of using `defineExpose`? 

This would actually make public methods accessible through `ref.value?.$.exposeProxy` isolated, meaning without props or other things that are not related to public methods. 

And `ref.value?.publicMethod()` would still be available.

**Comment by hinogi** [comment]
![image](https://github.com/quasarframework/quasar/assets/4602609/6e923e0e-47a0-457b-a898-62921ac6d1fe)
is this what you mean?

**Reply by hinogi** [reply]
Only if setup flag exposed or defineExpose is used do you have expose proxy as far as I understand it from this test 

https://github.com/vuejs/core/blob/ae97e5053895eeaaa443306e72cd8f45da001179/packages/compiler-sfc/__tests__/compileScript.spec.ts#L376

**Reply by Dtsiantaris** [reply]
Yes. But this code wouldn't work right?
It would return `instance.proxy` because `instance.exposed` would be undefined.
As you said `exposeProxy` (and `instance.exposed` I would add) are defined when `setup expose` or `defineExpose` macro are used.
